pandas/tests/tools/test_to_datetime.py

```diff
@@ -1123,17 +1123,32 @@ def test_iso8601_strings_mixed_offsets_with_naive(self):
 
     def test_mixed_offsets_with_native_datetime_raises(self):
         # GH 25978
-        ser = Series(
+
+        vals = [
+            "nan",
+            Timestamp("1990-01-01"),
+            "2015-03-14T16:15:14.123-08:00",
+            "2019-03-04T21:56:32.620-07:00",
+            None,
+        ]
+        ser = Series(vals)
+        assert all(ser[i] is vals[i] for i in range(len(vals)))  # GH#40111
+
+        mixed = to_datetime(ser)
+        expected = Series(
             [
-                "nan",
+                "NaT",
                 Timestamp("1990-01-01"),
-                "2015-03-14T16:15:14.123-08:00",
-                "2019-03-04T21:56:32.620-07:00",
+                Timestamp("2015-03-14T16:15:14.123-08:00").to_pydatetime(),
+                Timestamp("2019-03-04T21:56:32.620-07:00").to_pydatetime(),
                 None,
-            ]
+            ],
+            dtype=object,
         )
+        tm.assert_series_equal(mixed, expected)
```

Comment:
does this PR fully close the issue? The issue refers to this as not being idempotent.

Reply:
can keep it open with a comment that the idempotency has not been addressed

Reply:
sgtm